[DebugInfo][RemoveDIs] Delete debug-info-format flag

jmorse opened the pull request:
This flag was used to let us incrementally introduce debug intrinsics into LLVM, however everything is now using intrinsics. It serves no purpose now, so delete it.

jmorse committed:
[DebugInfo][RemoveDIs] Delete debug-info-format flag

This flag was used to let us incrementally introduce debug intrinsics into
LLVM, however everything is now using intrinsics. It serves no purpose now,
so delete it.

jmorse committed:
MLIR: unconditionally convert from #dbg_records

(Because they're on everywhere, so we don't need to check the flag)

OCHyams reviewed (approved):
nit: Commit message refers to intrinsics instead of records?

Pls appease clang-format-bot

LGTM

jmorse committed:
A further MLIR debug-info change

There's no need to update the flag now that it's going; but place a call to
convert-to-new-dbg-values where we used to call the set-the-flag function.
This is a boundary where we might convert intrinsics into records -- I
suspect that if MLIR is using DIBuilder, it's already creating records
anyway.

jmorse committed:
Peace with clang-format
